src/main/java/hudson/plugins/clover/CloverPublisher.java

```diff
@@ -224,6 +280,54 @@ public String invoke(File file, VirtualChannel virtualChannel) throws IOExceptio
         }
     }
 
+    private void processCloverXml(Run<?, ?> build, FilePath workspace, TaskListener listener, FilePath
+            coverageReport, FilePath buildTarget) throws InterruptedException {
+
+        String workspacePath = "";
+        try {
+            workspacePath = workspace.act(new SlaveToMasterFileCallable<String>() {
+                public String invoke(File file, VirtualChannel virtualChannel) throws IOException {
+                    try {
+                        return file.getCanonicalPath();
+                    } catch (IOException e) {
+                        return file.getAbsolutePath();
+                    }
+                }
+            });
+        } catch (IOException e) {
+        }
+        if (!workspacePath.endsWith(File.separator)) {
+            workspacePath += File.separator;
+        }
+
+        final File cloverXmlReport = getCloverXmlReport(build);
+        if (cloverXmlReport.exists()) {
+            listener.getLogger().println("Publishing Clover coverage results...");
+            ProjectCoverage result = null;
+            try {
+                result = CloverCoverageParser.parse(cloverXmlReport, workspacePath);
+            } catch (IOException e) {
+                Util.displayIOException(e, listener);
+                e.printStackTrace(listener.fatalError("Unable to copy coverage from " + coverageReport + " to " + buildTarget));
+                build.setResult(Result.FAILURE);
```

Comment:
Should we put build as failed in case of couldn't parse a coverage file? Maybe see, simply I am wondering if we could add an error message in the report and that's all. Or maybe configure if we want to make build fail in these cases. Also I am thinking that probably this is a very strange error so we can leave as is.